internal/controller/agenttags/tags.go

```diff
@@ -82,20 +82,24 @@ func LabelsFromTags(tags []string) (map[string]string, []error) {
 	return labels, append(errs1, errs2...)
 }
 
-// JobTagsMatchAgentTags reports whether each tag key in `jobTags` is also
-// present in `agentTags`, and the tag value in `jobTags` is either "*" or the
+// AgentTagsMatchJobTags reports whether each tag key in `agentTags` is also
+// present in `jobTags`, and the tag value in `jobTags` is either "*" or the
 // same as the tag value in `agentTags`.
 //
 // In the future, this may be expanded to: if the tag value `agentTags` is in some
 // set of strings defined by the tag value in `jobTags` (eg a glob or regex)
 // See https://buildkite.com/docs/agent/v3/cli-start#agent-targeting
-func JobTagsMatchAgentTags(jobTags, agentTags map[string]string) bool {
-	for k, v := range jobTags {
-		agentTagValue, exists := agentTags[k]
+
+func AgentTagsMatchJobTags(agentTags, jobTags map[string]string) bool {
```

Comment:
Given that this is in the `agenttags` package, `AgentTags` could be dropped from the start of the name so that calling it reads like `agenttags.MatchJobTags(...)`:

```suggestion
func MatchJobTags(agentTags, jobTags map[string]string) bool {
```